moped-editor/src/views/projects/projectView/ProjectComponents/ComponentDrawTools.js

```diff
@@ -0,0 +1,142 @@
+import React, { useRef } from "react";
+import ComponentsDrawControl from "src/components/Maps/ComponentsDrawControl";
+import { makeDrawnFeature } from "./featureUtils";
+import { cloneDeep } from "lodash";
+
+/**
+ * Renders project component draw tools
+ * @param {Function} setDraftComponent - function to update the draft component
+ * @param {String} linkMode - tracks if we are editing "lines" or "points"
+ * @param {Function} setCursor - function to update the cursor type
+ * @param {Function} setIsDrawing - function to update if we are drawing or not
+ * @returns {JSX.Element}
+ */
+const ComponentDrawTools = ({
+  setDraftComponent,
+  linkMode,
+  setCursor,
+  setIsDrawing,
+}) => {
+  const drawControlsRef = useRef();
+  const shouldShowDrawControls = linkMode === "points" || linkMode === "lines";
+
+  const onCreate = ({ features: createdFeaturesArray }) => {
+    setDraftComponent((prevDraftComponent) => {
```

Comment:
Not sure if I'm a huge fan of nesting all of this logic inside this state setter (same in `onUpdate` and `onDelete` below). I left them for now, but I'm going to think on it.

Reply:
there's some repetition, but it's also very easy to track what's happening and i'm cool with optimizing later, especially if we're not super confident this is the final iteration of the draw behavior.

i'm not feeling very creating at the moment, but one way to improve readability a bit could be to wrap these handlers in `setDraftComponent` ?

```javascript
onCreate={(e) => setDraftComponent(onCreate(e)}
```

not sure that's actually an improvement 🤪

Reply:
Yep, I'm on the same page. I think that updating with some helpers is another way to handle this, but I think that I'll leave it this way for now. Punt! 🏈